package maidsafe.dht.protobuf;

message Endpoint {
  required bytes ip = 1;
  required int32 port = 2;
}

message Contact {
  optional Endpoint endpoint = 1;
  repeated bytes local_ips = 2;
  optional int32 local_port = 3;
  optional Endpoint rendezvous = 4;
  optional bool tcp443 = 5;
  optional bool tcp80 = 6;
  optional bool prefer_local = 7;
  required bytes node_id = 8;
  optional bytes public_key_id = 9;
  optional bytes public_key = 10;
  optional bytes other_info = 11;
}

message BootstrapContacts {
  repeated Contact contact = 1;
}